lib/registry.py

```diff
@@ -557,6 +559,11 @@ def request(self, method, url, statuses={200}, out=None, **kwargs):
          out.start(length)
          for chunk in res.iter_content(ch.HTTP_CHUNK_SIZE):
             out.write(chunk)
+            m.update(chunk)
+         # Check for integrity of downloaded blob by validating its hash digest against expected hash digest
+         if (expdigest is not None):
+            if (expdigest != m.hexdigest()):
```

Comment:
This can be condensed to a single if statement:

```suggestion
         if (expdigest is not None and expdigest != m.hexdigest()):
```